miner/worker.go

```diff
@@ -1095,8 +1128,15 @@ func (w *worker) generateWork(genParams *generateParams) *newPayloadResult {
 		err := w.fillTransactions(interrupt, work)
 		if errors.Is(err, errBlockInterruptedByTimeout) {
 			log.Warn("Block building is interrupted", "allowance", common.PrettyDuration(w.newpayloadTimeout))
+		} else if errors.Is(err, errBlockInterruptedByResolve) {
+			timer.Stop() // don't need timeout interruption any more
```

Comment:
Maybe it's better to just always call `timer.Stop()` after `fillTransactions`, to cover the success-case before timeout also? And then remove the `defer timer.Stop()`

Reply:
Done. Wanted to keep the diff minimal, but that also works and is cleaner.